src/table-semantic/table-builder-column.js

```diff
@@ -0,0 +1,16 @@
+/*
+Copyright (c) 2018-2019 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+
+import type {TableBuilderColumnPropsT} from './types.js';
+
+export default class TableBuilderColumn<T> extends React.Component<
```

Comment:
What do you think are the trade-offs between this pattern versus a `columns` prop on the parent component? Why did you choose this approach?

Reply:
I wanted to provide a declarative means of defining columns. Using an array of configuration feels out of place when composing a React component and scatters the rendering logic throughout the component.

Consider the common case of rendering a link in a table:

```js
function Users({ data }) {
  return (
    <TableBuilder data={data}>
      <TableBuilderColumn header="Name" sortable>
         {user => <a href={user.profileUrl}>{user.name}</a>}
      </TableBuilderColumn>
      <TableBuilderColumn header="Age" numeric sortable>
         {user => user.age}
      </TableBuilderColumn>
    </TableBuilderColumn>
  );
}
```

vs

```js
function Users({ data }) {
  const columns = [
    {
      header: 'Name',
      renderCell: user => <a href={user.profileUrl}>{user.name}</a>,
      isSortable: true,
    },
    {
      header: 'Age',
      renderCell: user => user.age,
      isNumeric: true,
      isSortable: true,
    },
  ];

  return <TableBuilder columns={columns} data={data} />;
}
```

Reply:
Also, I believe providing the component api provides the best of both worlds as you can still rather easily use an array if that's your preference:

```jsx
function Users({ data }) {
  const columns = [
    {
      header: 'Name',
      children: user => <a href={user.profileUrl}>{user.name}</a>,
      sortable: true,
    },
    {
      header: 'Age',
      children: user => user.age,
      numeric: true,
      sortable: true,
    },
  ];

  return (
    <TableBuilder data={data}>
      {columns.map(col => <TableBuilderColumn {...col} />)}
    </TableBuilder>
  );
}
```

Reply:
I can agree with the desire to consolidate rendering logic. My concern is that it diverges from how the 'basic' table component consumes an array of column titles. Your last example seems like a reasonable option for anyone prefers an array-based approach. Let's check in on the API before 'stabilizing' the component

Reply:
I totally get that. I thought about not having a columns attribute on the basic table at all, but have a header boolean which assumes the first row of the data set IS the headers like a CSV file. But I wanted to keep the basic api of table-semantic 1:1 with the div-based basic table.

The divergent API of table-builder is the reason I created the additional components instead of baking it all into a single component.